app/services/ArrivalMessageSummaryService.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import cats.data.NonEmptyList
+import models.MessageType._
+import models.Arrival
+import models.MessageId
+import models.MovementMessage
+import models.MovementMessageWithStatus
+import models.MovementMessageWithoutStatus
+
+class ArrivalMessageSummaryService {
+
+  def arrivalNotification(arrival: Arrival): (MovementMessage, MessageId) = arrival.messagesWithId match {
+    case NonEmptyList(arrivalNotification, Nil)                                                     => arrivalNotification
+    case NonEmptyList(arrivalNotification, _ :: Nil)                                                => arrivalNotification
+    case NonEmptyList((msg @ MovementMessageWithStatus(_, ArrivalNotification, _, _, _), id), tail) =>
+      // This is a workaround since we cannot infer the type of head
+      // to be (MovementMessageWithStatus, MessageId) using @ in the pattern match
+      val head: (MovementMessageWithStatus, MessageId) = (msg, id)
+
+      tail
+        .foldLeft(NonEmptyList.of(head))({
+          case (acc, (m @ MovementMessageWithStatus(_, ArrivalNotification, _, _, _), mid)) => acc :+ Tuple2(m, mid)
+          case (acc, _)                                                                     => acc
+        })
+        .toList
+        .maxBy(_._1.messageCorrelationId)
+
+    case _ => ??? // Unreachable but unprovable
```

Comment:
even if its unreachable shouldn't it do something?

Reply:
Throwing an exception would be the only other thing, but the question is what sort of exception?

Possibly a RuntimeException, but that doesn't seem informative enough, it more that we hit a scenario that we shouldn't ever be able to get to 😕 

Any suggestions?

Reply:
The only thing we can do here would possibly be to throw an exception (until someone makes this better).

I'm not sure a RuntimeException is correct here 🤔 we want the exception to indicate that we've hit a scenario that we shouldn't be able to. Any suggestions?